frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -32,26 +47,42 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
         apiVersion: 'v1',
         propagationPolicy: 'Foreground',
       };
-      handlePromise(
-        Promise.all([
-          k8sKill(SubscriptionModel, subscription, {}, deleteOptions),
-          ...(subscription?.status?.installedCSV
-            ? [
-                k8sKill(
-                  ClusterServiceVersionModel,
-                  {
-                    metadata: {
-                      name: subscription.status.installedCSV,
-                      namespace: subscription.metadata.namespace,
-                    },
+      const k8kills = [
+        k8sKill(SubscriptionModel, subscription, {}, deleteOptions),
+        ...(subscription?.status?.installedCSV
+          ? [
+              k8sKill(
+                ClusterServiceVersionModel,
+                {
+                  metadata: {
+                    name: subscription.status.installedCSV,
+                    namespace: subscription.metadata.namespace,
                   },
-                  {},
-                  deleteOptions,
-                ).catch(() => Promise.resolve()),
-              ]
-            : []),
-        ]),
-      )
+                },
+                {},
+                deleteOptions,
+              ).catch(() => Promise.resolve()),
+            ]
+          : []),
+      ];
+      if (deleteAllOperands && operands) {
+        operands.forEach((operand: K8sResourceKind) => {
```

Comment:
Prefer `K8sResourceCommon` which has stronger type safety.

```suggestion
        operands.forEach((operand: K8sResourceCommon) => {
```

Reply:
fixed